### DISCUSSION TITLE
Can Cypress simulate a user browsering with "ad-blocker" enabled?

### DISCUSSION THREAD

**Discussion by darrinmn9** [cypress-io/cypress#29002] [Questions and Help] [answered]
Ad-blockers are so prevalent today, with some browsers even having them built in. This can add complications, especially with websites that surface cookie consent forms (as some legal jurisdictions require explicit consent to continue). Is there a way in cypress to "simulate an ad-blocker", or do something like, "Block any and all assets from the domain `abc.com`"? 

I realize this can sorta be done for http requests via `cy.intercept`, but I am hoping for a way to be able to simulate the blocking of all assets, including scripts/css/images/ajax/etc, and to be able to run tests that a assert the website is still functional in this situation.

**Comment by portyu9** [comment, marked as answer]
Hi @darrinmn9 

Yes. For this use case, use Cypress's `blockHosts` configuration rather than individual `cy.intercept()` calls.

For example:

```js
it(
  'works when abc.com is blocked',
  {
    blockHosts: [
      'abc.com',
      '*.abc.com',
    ],
  },
  () => {
    cy.visit('/')
  }
)
```

That blocks requests to those hosts, including resources such as scripts, stylesheets, images, and API requests, which is much closer to the scenario you described.

You can also configure `blockHosts` globally in `cypress.config.js` / `cypress.config.ts`.

One current-version detail: starting with Cypress 16, do **not** call `Cypress.config('blockHosts', ...)` from inside a running test. Use test/suite-specific configuration as above, or configure it before the test starts.

Please mark this answer as accepted if it helped, thank you.
